core/fifo_cache.go

```diff
@@ -18,10 +20,16 @@ type FIFOCache[K comparable, V any] struct {
 	m      map[K]V
 }
 
+// NewFIFOCache creates a new First-In-First-Out cache of size [limit].
+//
+// If [limit] is less than 16, [limit] will be overwritten with 16.
```

Comment:
Alternatively, we could have a no-op cache if 0 is passed in (which occurs in some tests).